libs/blocks/library-config/library-config.js

```diff
@@ -1,192 +1,137 @@
-      blocks.forEach((block) => {
-        const item = document.createElement('li');
-        const name = document.createElement('p');
-        name.textContent = getAuthorName(block) || getBlockName(block);
-        const copy = document.createElement('button');
-        copy.addEventListener('click', (e) => {
-          const table = getTable(block, getBlockName(block), blockGroup.value);
-          e.target.classList.add('copied');
-          setTimeout(() => { e.target.classList.remove('copied'); }, 3000);
-          const blob = new Blob([table], { type: 'text/html' });
-          createCopy(blob);
-        });
-        item.append(name, copy);
-        list.append(item);
-      });
-    }
-  };
+async function loadBlocks(content, list) {
+  const { default: blocks } = await import('./lists/blocks.js');
+  blocks(content, list);
 }
 
-function stub(list) {
-  const item = document.createElement('li');
-  const name = document.createElement('p');
-  name.textContent = 'This library isn\'t supported';
-  item.append(name);
-  list.append(item);
+async function loadPlaceholders(content, list) {
+  const { default: placeholders } = await import('./lists/placeholders.js');
+  placeholders(content, list);
 }
 
-async function loadList(type, list) {
+async function loadList(type, content, list) {
   list.innerHTML = '';
-  switch (type.text) {
-    case 'Blocks':
-      loadBlockList(type.paths, list);
+  switch (type) {
+    case 'blocks':
+      loadBlocks(content, list);
       break;
-    case 'Templates':
-      stub(list);
-      break;
-    case 'Placeholders':
-      stub(list);
-      break;
-    case 'Tokens':
-      stub(list);
+    case 'placeholders':
+      loadPlaceholders(content, list);
       break;
     default:
       await import('../../utils/lana.js');
       window.lana.log(`Library type not supported: ${type}`, { clientId: 'milo', sampleRate: 100 });
   }
 }
 
-export default function init(el) {
-  const librarEls = el.querySelectorAll('a');
-
-  const blockPaths = [...librarEls].map((lib) => { return lib.href; });
-
-  const { searchParams: sp } = new URL(window.location.href);
-  const additionalBlockLibs = sp.getAll('blocks')
-    .map((lib) => {
-      if (lib.startsWith('http')) return lib;
-      return `https://${lib}.hlx.page/docs/library/blocks.json`;
-    });
-  blockPaths.push(...additionalBlockLibs);
-
-  const libraries = [ { text: 'Blocks', paths: blockPaths } ];
+async function fetchLibrary(domain) {
+  const resp = await fetch(`${domain}${LIBRARY_PATH}`);
+  if (!resp.ok) return;
+  const json = await resp.json();
+  Object.keys(json).forEach((key) => {
+    if (key.startsWith(':')) delete json[key];
+  });
+  return json;
+}
 
-  el.querySelector('div').remove();
+async function getSuppliedLibrary() {
+  const url = new URL(window.location.href);
+  const library = url.searchParams.get('library');
+  if (!library) return null;
+  return await fetchLibrary(`https://main--${library}.hlx.live`);
```

Comment:
Add support for `passConfig` (https://github.com/adobe/helix-sidekick-extension/pull/168)

Also I _think_ that the `await` can be removed when returning.

```suggestion
  const library = url.searchParams.get('library');
  if (library) return await fetchLibrary(`https://main--${library}.hlx.live`);
  const repo = url.searchParams.get('repo');
  const owner = url.searchParams.get('owner');
  if (repo && owner) return await fetchLibrary(`https://main--${repo}--${owner}.hlx.live`
```

Reply:
I went with...

```js
async function getSuppliedLibrary() {
  const { searchParams } = new URL(window.location.href);
  const repo = searchParams.get('repo');
  const owner = searchParams.get('owner');
  if (!repo && !owner) return null;
  return fetchLibrary(`https://main--${repo}--${owner}.hlx.live`);
}
```